docs/developer/architecture/core/logging-service.asciidoc

```diff
@@ -0,0 +1,545 @@
+The main idea is to have consistent logging behaviour (configuration, log format etc.) across the entire Elastic Stack where possible.
+
+=== Loggers, Appenders and Layouts
+
+{kib} logging system has three main components: _loggers_, _appenders_ and _layouts_. These components allow us to log
+messages according to message type and level, to control how these messages are formatted and where the final logs
+will be displayed or stored.
+
+__Loggers__ define what logging settings should be applied to a particular logger.
+
+__<<logging-appenders,Appenders>>__ define where log messages are displayed (eg. stdout or console) and stored (eg. file on the disk).
+
+__<<logging-layouts,Layouts>>__ define how log messages are formatted and what type of information they include.
+
+[[logger-hierarchy]]
+=== Logger hierarchy
+
+Every logger has a unique name that follows a hierarchical naming rule. The logger is considered to be an
+ancestor of another logger if its name followed by a `.` is a prefix of the descendant logger. For example, a logger
+named `a.b` is an ancestor of logger `a.b.c`. All top-level loggers are descendants of a special `root` logger at the top of the logger hierarchy. The `root` logger always exists and
+fully configured.
+
+You can configure _<<log-level, log level>>_ and _appenders_ for a specific logger. If a logger only has a _log level_ configured, then the _appenders_ configuration applied to the logger is inherited from the ancestor logger.
+
+NOTE: In the current implementation we __don't support__ so called _appender additivity_ when log messages are forwarded to _every_ distinct appender within
+ancestor chain including `root`. That means that log messages are only forwarded to appenders that are configured for a particular logger. If a logger doesn't have any appenders configured, the configuration of that particular logger will be inherited from its closest ancestor. 
+
+[[log-level]]
+=== Log level
+
+Currently we support the following log levels: _all_, _fatal_, _error_, _warn_, _info_, _debug_, _trace_, _off_.
+
+Levels are ordered, so _all_ > _fatal_ > _error_ > _warn_ > _info_ > _debug_ > _trace_ > _off_.
+
+A log record is being logged by the logger if its level is higher than or equal to the level of its logger. Otherwise,
+the log record is ignored.
+
+The _all_ and _off_ levels can be used only in configuration and are just handy shortcuts that allow you to log every
+log record or disable logging entirely  or for a specific logger. These levels are also configurable as <<logging-cli-migration,cli arguments>>.
+
+
+[[logging-layouts]]
+=== Layouts
+
+Every appender should know exactly how to format log messages before they are written to the console or file on the disk.
+This behaviour is controlled by the layouts and configured through `appender.layout` configuration property for every
```

Comment:
```suggestion
This behaviour is controlled by the layouts and configured through `appender.layout` configuration property for every
```

Also same question `behaviour` or `behavior`

Reply:
But I like my "u"s! I've changed it but FWIW, "behavior" is spelt "behaviour" [everywhere else](https://internet.psych.wisc.edu/wp-content/uploads/532-Master/532-UnitPages/Unit-13/Grammarist_Behavior_NoDate.pdf) in the word besides the US 😉

Reply:
Yeah I promise I'm not arbitrarily trying to get rid of those `u`s! 😄  

Mostly I'm just wondering if we are supposed to prefer one or the other in our docs as a convention (TBH I don't know the answer to this, but figured the docs team would have a stance on it)